materialize-databricks/sqlgen.go

```diff
@@ -50,12 +50,34 @@ var databricksDialect = func() sql.Dialect {
 	// https://docs.databricks.com/en/sql/language-manual/sql-ref-json-path-expression.html
 	var jsonMapper = sql.NewStaticMapper("STRING", sql.WithElementConverter(jsonConverter))
 
+	var customDDLMapper = func(mapper sql.TypeMapper) sql.FieldConfigMapper {
+		return sql.FieldConfigMapper{
+			Delegate: mapper,
+			Map: func(m *sql.MappedType, config json.RawMessage) error {
+				type customDDLFieldConfig struct {
+					DDL string `json:"DDL,omitempty"`
+				}
+
+				var fieldConfig customDDLFieldConfig
+				if err := json.Unmarshal(config, &fieldConfig); err != nil {
+					return err
+				}
+
+				if fieldConfig.DDL != "" {
+					m.DDL = fieldConfig.DDL
+				}
+
+				return nil
+			},
+		}
+	}
+
 	// https://docs.databricks.com/en/sql/language-manual/sql-ref-datatypes.html
 	var mapper sql.TypeMapper = sql.ProjectionTypeMapper{
 		sql.ARRAY:    jsonMapper,
 		sql.BINARY:   sql.NewStaticMapper("BINARY"),
 		sql.BOOLEAN:  sql.NewStaticMapper("BOOLEAN"),
-		sql.INTEGER:  sql.NewStaticMapper("BIGINT"),
+		sql.INTEGER:  customDDLMapper(sql.NewStaticMapper("BIGINT")),
```

Comment:
At the moment I have only added this mapper to integers, just for the sake of "this will be hard to take back, so let's not give it away prematurely"